packages/core/utils/src/set-creator-fields.ts

```diff
@@ -12,8 +12,8 @@ interface User {
   id: string | number;
 }
 
-export default ({ user, isEdition = false }: Options) =>
-  (data: object) => {
+export default <T extends object>({ user, isEdition = false }: Options) =>
```

Comment:
Shouldn't the generic be attached to the returned function instead? Or is it made on prupose?

Reply:
T is used for the returned function type only (see line below), that's what I meant
`  (data: T) => {`

Reply:
Since it's made to be curried, I think it makes sense